Adding Kosh python package v2.0

doutriaux1 opened the pull request.

doutriaux1 committed:
Adding Kosh python package v2.0

doutriaux1 committed:
flake8

adamjstewart commented on var/spack/repos/builtin/packages/py-kosh/package.py:
```diff
@@ -0,0 +1,28 @@
+
+class PyKosh(PythonPackage):
+    """Kosh allows codes to store, query, share data via an easy-to-use Python API.
+Kosh lies on top of Sina and as a result can use any database backend supported by Sina.
+In adition Kosh aims to make data access and sharing as simple as possible.
+"""
```
I would indent this 4 spaces

adamjstewart replied:
```suggestion
    Kosh lies on top of Sina and as a result can use any database backend supported by Sina.
    In adition Kosh aims to make data access and sharing as simple as possible.
    """
```
May need to wrap some of these lines if they are too long.

adamjstewart commented on var/spack/repos/builtin/packages/py-kosh/package.py:
```diff
@@ -0,0 +1,28 @@
+"""
+
+    homepage = "https://github.com/LLNL/kosh"
+    git = "https://github.com/LLNL/kosh.git"
```
It's better to fetch from a URL so that we can checksum each version

adamjstewart replied:
You're going to need a `url` variable, this won't build as is

adamjstewart commented on var/spack/repos/builtin/packages/py-kosh/package.py:
```diff
@@ -0,0 +1,28 @@
+    version('2.0', tag="v2.0")
+
+    depends_on('py-setuptools', type='build')
+    depends_on("py-sina", type=("build", "run"))
```
```suggestion
    depends_on("py-llnl-sina@1.11:", type=("build", "run"))
```
It looks like the `py-sina` package should be renamed to `py-llnl-sina`. There are different packages on PyPI: (https://pypi.org/project/llnl-sina/, https://pypi.org/project/sina/). If you're concerned about users using the wrong name, you can duplicate the package and deprecate the old one.

doutriaux1 replied:
@adamjstewart can I do this in this PR? Or do you prefer me to submit a separate PR for `py-sina` renaming first?

adamjstewart replied:
Either is fine.

doutriaux1 committed:
renamed py-sina to py-llnl-sina

I'm not worried about keep py-sina around since for one
we don't have any outside llnl users and I don't think any of them
used sina via spack yet

Also using sha256 rather than tag for Kosh

doutriaux1 committed:
Merge remote-tracking branch 'origin/develop' into py-kosh

doutriaux1 committed:
put amintainers on one line mostly to retrigger builds

doutriaux1 commented:
@adamjstewart I have no idea why the style fails... The error message is not clear to me either.

doutriaux1 committed:
renamed pysina class added url for kosh

adamjstewart commented on var/spack/repos/builtin/packages/py-kosh/package.py:
```diff
@@ -0,0 +1,27 @@
+    url = "https://github.com/LLNL/kosh/archive/refs/tags/v2.0.tar.gz"
+
+    # notify when the package is updated.
+    maintainers = ['doutriaux1', ]
```
```suggestion
    maintainers = ['doutriaux1']
```

doutriaux1 committed:
adressing @adamjstewart comments

adamjstewart commented on var/spack/repos/builtin/packages/py-kosh/package.py:
```diff
@@ -0,0 +1,28 @@
+
+
+class PyKosh(PythonPackage):
+    """i
```
```suggestion
    """
```

adamjstewart commented on var/spack/repos/builtin/packages/py-kosh/package.py:
```diff
@@ -0,0 +1,28 @@
+    version('2.0', sha256='059e431e3d3219b53956cb464d9e10933ca141dc89662f55d9c633e35c8b3a1e')
+
+    depends_on('py-setuptools', type='build')
+    depends_on("py-llnl-sina", type=("build", "run"))
```
```suggestion
    depends_on("py-llnl-sina@1.11:", type=("build", "run"))
```

doutriaux1 committed:
looks like these are the last tweaks/typos. Thanks @adamjstewart
